### PR TITLE
Scope data by keys

### PR DESCRIPTION
**Pull request by almirsarajcic**
Replaces #113.

### CODE REVIEW

File: lib/chat/db/scope/key_scope.ex [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      |> Enum.map(fn {:file_index, _reader_hash, file_key, _message_id} -> file_key end)
+      |> MapSet.new()
+
+    file_chunks =
+      snap
+      |> db_keys_stream({:file_chunk, 0, 0, 0}, {:"file_chunk\0", 0, 0, 0})
+      |> Stream.filter(fn {:file_chunk, file_key, _chunk_start, _chunk_end} ->
+        MapSet.member?(file_keys, file_key)
+      end)
+      |> MapSet.new()
```
Comment (sergey-lukianov) [inline lib/chat/db/scope/key_scope.ex:91]: db should not have `file_chunks` by now. 
But we need to produce them for files (from fileFs) to be copied.
We can do so from `chunk_keys`

File: lib/chat/db/scope/key_scope.ex [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      |> Enum.map(fn {:memo_index, _reader_hash, key} -> key end)
+      |> MapSet.new()
+
+    memos =
+      snap
+      |> db_keys_stream({:memo, 0}, {:"memo\0", 0})
+      |> Stream.filter(fn {:memo, key} ->
+        MapSet.member?(memo_keys, key)
+      end)
+      |> MapSet.new()
```
Comment (sergey-lukianov) [inline lib/chat/db/scope/key_scope.ex:131]: This looks like a pattern shared with `room_invites` at least. May we segregate it in a function?

**Review by sergey-lukianov — COMMENTED** [review]
and you can skip copying test if see no obvious reason for it to fail. It was unstable for me (rerun helped to make it pass)

File: lib/chat/db/scope/key_scope.ex (hunk 2 above)
Comment (almirsarajcic) [reply lib/chat/db/scope/key_scope.ex:131]: Sure.

File: lib/chat/db/scope/key_scope.ex (hunk 1 above)
Comment (almirsarajcic) [reply lib/chat/db/scope/key_scope.ex:91]: That's right. I realized that while I was writing the tests that it's just a convenient API for the file system and that there's no DB for that. I'll remove this part of the code.

**Review by sergey-lukianov — APPROVED** [review]

**Pull request merged**